### PR TITLE
[trace-view] Added support for tracking struct/enum values

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR adds support for tracking variable values for structs and enums. It also does some code cleanup (mostly via renamings)

## Test plan 

Tested manually

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts [hunk 1]
```diff
@@ -3,104 +3,125 @@
 
 import * as fs from 'fs';
 import { FRAME_LIFETIME, ModuleInfo } from './utils';
+import { IRuntimeCompundValue, RuntimeValueType } from './runtime';
+
 
 // Data types corresponding to trace file JSON schema.
 
-interface ITraceModule {
+interface JSONTraceModule {
```
Comment (awelc) [inline external-crates/move/crates/move-analyzer/trace-adapter/src/trace_utils.ts:11]: These renamings are unrelated to the main purpose of this PR but things were getting too confusing with the old naming scheme

File: external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts [hunk 2]
```diff
@@ -268,41 +272,92 @@ export class MoveDebugSession extends LoggingDebugSession {
...
+            return {
+                name,
+                type,
+                value: '(' + value.length + ')[...]',
+                variablesReference: compoundValueReference
+            };
+        } else {
+            const compoundValueReference = this.variableHandles.create(value);
+            const accessChainParts = value.type.split('::');
+            const datatypeName = accessChainParts[accessChainParts.length - 1];
```
Comment (tzakian) [inline external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts:306]: Wondering if the `datatypeName` should include the module name as well? Although by that logic we should include the address as well and display the fully qualified typename which isn't really practical...

Not needed now, but more something to think about for the future probably is if there is any way to provide a "mouse-over" name here (or a expandable name) where you could get the fully-qualified typename of the type if you want?

**Review by tzakian — APPROVED** [review]
Looks good to me! Just one comment about disambiguating typenames for the future, but this is 100% fine to go in as-is.

File: external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts (hunk 2 above)
Comment (awelc) [reply external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts:306]: It'a a great idea actually! I guess, I did not think about it enough... In any case, I added support for on-hover showing the fully qualified type for non-primitive values. Here are some examples:
image

image

image

I also added support for the same info being displayed for inner fields of structs/enums:
image

Comment (tzakian) [reply external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts:306]: Amazing! That's awesome :)

**Pull request merged**